src/error.rs

```diff
@@ -151,6 +151,9 @@ pub enum ValidationErrorKind {
     UniqueItems,
     /// Reference contains unknown scheme.
     UnknownReferenceScheme { scheme: String },
+
+    /// Unexpected error. This usually represent a bug into the validation
+    Unexpected { validator_representation: String },
```

Comment:
The idea of having the `ValidationError::Unexpected` is to enable the definition of default `Validate::validate_*` methods which is very handy as it avoids a massive code duplication and keep binary size under control.

A possible alternative would be to use `panic!` or `unimplemented!` but it feels like "wrong" to abort the program for this reason.

So I originally felt that it was an acceptable middle-ground solution to have this "catch-all" validation error; understanding that it is the reflection of a bug it provides to the user such information and suggests to report it to the library such that can be fixed ASAP.

Please let me know what are your thoughts about this and if you need/want me to remove all the default implementations of `Validate::validate_`.

Reply:
Oh, I didn't see it initially, so I agree that this might be an acceptable middle-ground

Reply:
And I don't know how we can avoid this case statically :(

Reply:
I've tried re-iterating it again, looking for strategies to avoid this but nothing popped out to my eyes.

If you do agree we might have this merged and eventually opening an issue to try to remove this.